### PR TITLE
Added an Analysis#getDocumentOrDie(url: string) method.

### PR DESCRIPTION
**Pull request by usergenic**
- [ ] CHANGELOG.md has been updated
 - Added getDocumentOrDie() as a convenient option to just get a known analyzed document by url instead of dealing with multiple return type.

### CODE REVIEW

File: src/model/analysis.ts [hunk 1]
```diff
@@ -82,6 +88,22 @@ export class Analysis implements Queryable {
   }
 
   /**
+   * Simplified version of getDocument which simply throws an error when asked
+   * for a missing document.
+   */
+  getDocumentOrDie(url: string): Document {
```
Comment (justinfagnani) [inline src/model/analysis.ts]: `OrThrow`?

Comment (justinfagnani) [inline src/model/analysis.ts]: I think this should be viewed as temporary until we have getDocument return a better result class that makes it easier to handle warnings.

**Comment by justinfagnani** [comment]
@usergenic wdyt, should we close this and change the signature of getDocument() in 3.0 to return a better type?

**Comment by rictic** [comment]
Ooh! Yeah! We should totally do that for 3.0

**Comment by rictic** [comment]
Use a real `Result` type

**Comment by usergenic** [comment]
Closing this PR and created https://github.com/Polymer/polymer-analyzer/issues/772

**Pull request closed without merging**